src/lib/UI/Value/ValueFactory.php

```diff
@@ -180,17 +195,31 @@ public function createRelation(Relation $relation, Content $content): UIValue\Co
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
      * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
      */
     public function createRelationItem(RelationListItem $relationListItem, Content $content): UIValue\Content\Relation
     {
         $contentType = $content->getContentType();
         $relation = $relationListItem->getRelation();
         $fieldDefinition = $contentType->getFieldDefinition($relation->sourceFieldDefinitionIdentifier);
 
+        $contentInfo = $content->contentInfo;
+
+        try {
+            $relationLocation = $this->locationService->loadLocation($contentInfo->mainLocationId);
+        } catch (UnauthorizedException $e) {
+            // try different locations if main location is not accessible for the user
+            $relationLocations = $this->locationService->loadLocations($contentInfo);
+            if (empty($relationLocations)) {
+                throw $e;
+            }
+            $relationLocation = reset($relationLocations);
```

Comment:
`\eZ\Publish\Core\Repository\LocationService::loadLocations` guarantee that return value is `iterable` not an `array` (in 3.0). `reset` function shouldn't be used here, because code depends on concrete implementation instead of the contract.

Reply:
@adamwojs this PR is meant for v2.5, not v3.0 and the return value of `loadLocations` is array.